File: frontend/src/types/aiTutor.ts
export type TutorMode =
  | 'explain'
  | 'hint'
  | 'debug'
  | 'quiz_me'
  | 'interview_me'
  | 'review_code'
  | 'review_analysis'
  | 'explain_graph'
  | 'explain_error'
  | 'give_project'
  | 'create_practice'

export interface CreateSessionPayload {
  mode: TutorMode
  lesson_title?: string
  code?: string
  error_message?: string
  skill_level?: string
}

export interface TutorSession {
  id: string
  mode: TutorMode
  started_at: string
}

export interface ChatMessage {
  role: 'user' | 'assistant' | 'system'
  content: string
}

File: backend/app/schemas/ai_tutor.py
import uuid
from datetime import datetime

from pydantic import BaseModel, ConfigDict

from app.models.ai import TutorMode


class CreateSessionRequest(BaseModel):
    mode: TutorMode
    lesson_title: str | None = None
    code: str | None = None
    error_message: str | None = None
    skill_level: str | None = None


class SessionRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)
    id: uuid.UUID
    mode: TutorMode
    started_at: datetime


class MessageRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)
    id: uuid.UUID
    role: str
    content: str
    created_at: datetime
